### PR TITLE
connector/gitlab: Fix regexp in Link parser

### PR DESCRIPTION
**Pull request by lsjostro**
Fix regression in Link parser. 
.* is to ambiguous.

### CODE REVIEW

**Comment by ericchiang** [comment]
Can you add a test?

**Comment by lsjostro** [comment]
Added test

File: connector/gitlab/gitlab_test.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
+package gitlab
+
+import (
+	"testing"
+)
+
+func TestNextURL(t *testing.T) {
+	linkHeader := "<https://gitlab.com/api/v4/groups?page=3&per_page=20>; rel=\"next\", " +
```
Comment (ericchiang) [inline connector/gitlab/gitlab_test.go]: nit: we generally prefer table driven tests https://github.com/golang/go/wiki/TableDrivenTests

If you want to add that here that works, if not I'm fine with the test as is.

File: connector/gitlab/gitlab.go [hunk 2]
```diff
@@ -272,22 +269,31 @@ func (c *gitlabConnector) groups(ctx context.Context, client *http.Client) ([]st
...
+	}
+	return groups, nil
+}
 
-		if len(reNext.FindStringSubmatch(link)) > 1 {
-			apiURL = reNext.FindStringSubmatch(link)[1]
-		} else {
-			break
+func nextURL(url string, link string) string {
+	reNext := regexp.MustCompile("<([^>]+)>; rel=\"next\"")
```
Comment (ericchiang) [inline connector/gitlab/gitlab.go]: nit: move these to global vars so we don't re-compile them every time the function is called

```
var (
    reNext = regexp.MustCompile("...")
    reLast = regexp.MustCompile("...")
)

func nextURL() {
    // .. use reNext and reLast
}
```

File: connector/gitlab/gitlab.go [hunk 3]
```diff
@@ -272,22 +269,31 @@ func (c *gitlabConnector) groups(ctx context.Context, client *http.Client) ([]st
...
+		if url == lastPageURL {
+			return ""
 		}
+	} else {
+		return ""
+	}
 
+	if len(reNext.FindStringSubmatch(link)) > 1 {
+		return reNext.FindStringSubmatch(link)[1]
+	} else {
```
Comment (ericchiang) [inline connector/gitlab/gitlab.go]: go lint picked this up as fishy

```
/home/travis/gopath/src/github.com/coreos/dex/connector/gitlab/gitlab.go:296:9: if block ends with a return statement, so drop this else and outdent its block
Found 1 lint suggestions; failing.
```

File: connector/gitlab/gitlab_test.go (hunk 1 above)
Comment (lsjostro) [reply connector/gitlab/gitlab_test.go]: I'll fix! Thanks!

File: connector/gitlab/gitlab.go (hunk 2 above)
Comment (lsjostro) [reply connector/gitlab/gitlab.go]: Will do!

**Comment by lsjostro** [comment]
squashed and fixed the above comments

**Comment by lsjostro** [comment]
Awesome! Thanks! Looking forward to a patch release soon then. 
I have it hot patched in prod currently.

**Comment by lsjostro** [comment]
@ericchiang when can we expect this merged?

**Comment by ericchiang** [comment]
@lsjostro sorry was out of the office.

**Pull request merged**